common/src/main/java/com/tw/go/plugin/BuildStatusNotifierPlugin.java

```diff
@@ -188,8 +188,16 @@ GoPluginApiResponse handleStageNotification(GoPluginApiRequest goPluginApiReques
         return renderJSON(responseCode, response);
     }
 
-    private boolean isMaterialOfType(Map material, String pollerPluginId) {
-        return ((String) material.get("type")).equalsIgnoreCase("scm") && ((String) material.get("plugin-id")).equalsIgnoreCase(pollerPluginId);
+    private boolean isMaterialFromTypes(Map material, List<String> pollerPluginIds) {
+        if (((String) material.get("type")).equalsIgnoreCase("scm")) {
+            for (String pollerPluginId : pollerPluginIds) {
+                if(((String) material.get("plugin-id")).equalsIgnoreCase(pollerPluginId)) {
+                    return true;
+                }
+            }
```

Comment:
This could be refactored to use `contains` method on `List`.

Reply:
Fixed in https://github.com/gocd-contrib/gocd-build-status-notifier/pull/134/commits/df0a417717075a0519ceb21895fc567091afc758